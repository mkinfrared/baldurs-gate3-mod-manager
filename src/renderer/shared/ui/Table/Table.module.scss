.Table {
  --background: var(--secondary-8);
  --border-color: var(--secondary-7);

  @media (prefers-color-scheme: light) {
    --background: var(--secondary-2);
    --border-color: var(--secondary-3);
  }

  background-color: var(--background);
  border: 1px solid;
  border-collapse: collapse;
  border-color: var(--border-color);
  font-size: 16px;

  th,
  td {
    border: 1px solid var(--border-color);
    padding: 5px 10px;
  }
}
