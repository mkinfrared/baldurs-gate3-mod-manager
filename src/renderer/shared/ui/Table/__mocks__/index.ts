export { default } from "./Table.mock";
