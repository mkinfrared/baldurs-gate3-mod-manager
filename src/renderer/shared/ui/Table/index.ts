export { default } from "./Table";

export * from "./Table.type";
