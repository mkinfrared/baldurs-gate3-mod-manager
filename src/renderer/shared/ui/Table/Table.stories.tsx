import { Meta, StoryObj } from "@storybook/react";

import css from "./Story.module.scss";
import Table from "./Table";

type Story = StoryObj<typeof meta>;

const meta = {
  title: "UI/Table",
  component: Table,
  tags: ["autodocs"],
} satisfies Meta<typeof Table>;

const Default: Story = {
  args: {},
  render: (args) => (
    <div className={css.Story}>
      <Table {...args}>
        <tbody>
          <tr>
            <th colSpan={3}>Nissan</th>
          </tr>
          <tr>
            <th>Модель</th>
            <th>Комплектация</th>
            <th>Наличие</th>
          </tr>
          <tr>
            <td rowSpan={2}>Nissan Qashqai</td>
            <td>VISIA</td>
            <td>+</td>
          </tr>
          <tr>
            <td>TEKNA</td>
            <td>+</td>
          </tr>
          <tr>
            <td rowSpan={2}>Nissan X-Trail</td>
            <td>ACENTA</td>
            <td>+</td>
          </tr>
          <tr>
            <td>CONNECTA</td>
            <td>-</td>
          </tr>
        </tbody>
      </Table>
    </div>
  ),
};

export { Default };

export default meta;
