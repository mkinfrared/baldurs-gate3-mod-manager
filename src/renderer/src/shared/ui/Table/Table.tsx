import { classNames } from "@renderer/shared/lib/helpers";

import css from "./Table.module.scss";
import { TableProps } from "./Table.type";

const Table = ({ className, children }: TableProps) => (
  <table className={classNames(css.Table, className)} data-testid="Table">
    {children}
  </table>
);

export default Table;
