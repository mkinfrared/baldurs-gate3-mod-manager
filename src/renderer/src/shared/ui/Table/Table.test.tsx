import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import Table from "./Table";

describe("<Table />", () => {
  const Component = <Table />;

  it("should be defined", () => {
    expect(Table).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("Table");

    expect(element).toBeDefined();
  });
});
