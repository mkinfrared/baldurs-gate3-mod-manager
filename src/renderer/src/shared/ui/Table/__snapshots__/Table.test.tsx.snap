// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Table /> > should match the snapshot 1`] = `
<div>
  <table
    class="Table"
    data-testid="Table"
  />
</div>
`;
